File: src/main/scala/org/lolhens/minechanics/client/config/gui/element/LabelConfigEntry.scala
package org.lolhens.minechanics.client.config.gui.element

import cpw.mods.fml.client.config.GuiConfigEntries
import cpw.mods.fml.client.config.IConfigElement
import cpw.mods.fml.client.config.GuiConfig
import net.minecraft.util.EnumChatFormatting
import net.minecraft.client.renderer.Tessellator
import cpw.mods.fml.client.config.GuiConfigEntries.IConfigEntry

class LabelConfigEntry(owningScreen: GuiConfig, owningEntryList: GuiConfigEntries, configElement: IConfigElement[_]) extends ListEntryBase(owningScreen, owningEntryList, configElement) {
  override def drawEntry(slotIndex: Int, x: Int, y: Int, listWidth: Int, slotHeight: Int, tessellator: Tessellator, mouseX: Int, mouseY: Int, isSelected: Boolean): Unit = {
    if (!drawLabel) return
    val label = EnumChatFormatting.WHITE.toString() + name
    val drawX = owningScreen.entryList.labelX + (owningEntryList.scrollBarX - mc.fontRenderer.getStringWidth(label)) / 2
    val drawY = y + slotHeight / 2 - mc.fontRenderer.FONT_HEIGHT / 2
    mc.fontRenderer.drawString(label, drawX, drawY, 16777215)
  }
  override def keyTyped(c: Char, i: Int) = {}
  override def updateCursorCounter() = {}
  override def mouseClicked(i: Int, j: Int, k: Int) = {}
  override def isDefault() = true
  override def setToDefault() = {}
  override def isChanged() = false
  override def undoChanges() = {}
  override def saveConfigElement() = false
  override def getCurrentValues() = null
  override def getCurrentValue2() = null
}

File: src/main/java/org/lolhens/minechanics/core/config/gui/label/LabelConfigEntry.java
package org.lolhens.minechanics.core.config.gui.label;

import net.minecraft.client.renderer.Tessellator;
import net.minecraft.util.EnumChatFormatting;
import cpw.mods.fml.client.config.GuiConfig;
import cpw.mods.fml.client.config.GuiConfigEntries;
import cpw.mods.fml.client.config.GuiConfigEntries.ListEntryBase;
import cpw.mods.fml.client.config.IConfigElement;

public class LabelConfigEntry extends ListEntryBase {
    public LabelConfigEntry(GuiConfig owningScreen, GuiConfigEntries owningEntryList, IConfigElement<?> configElement) {
        super(owningScreen, owningEntryList, configElement);
    }

    @Override
    public void drawEntry(int slotIndex, int x, int y, int listWidth, int slotHeight, Tessellator tessellator, int mouseX,
            int mouseY, boolean isSelected) {
        if (drawLabel) {
            String label = EnumChatFormatting.WHITE.toString() + name;
            mc.fontRenderer.drawString(label,
                    owningScreen.entryList.labelX + (owningEntryList.scrollBarX - mc.fontRenderer.getStringWidth(label)) / 2, y
                            + slotHeight / 2 - mc.fontRenderer.FONT_HEIGHT / 2, 16777215);
        }
    }

    @Override
    public void keyTyped(char c, int i) {
    }

    @Override
    public void updateCursorCounter() {
    }

    @Override
    public void mouseClicked(int i, int j, int k) {
    }

    @Override
    public boolean isDefault() {
        return true;
    }

    @Override
    public void setToDefault() {
    }

    @Override
    public boolean isChanged() {
        return false;
    }

    @Override
    public void undoChanges() {
    }

    @Override
    public boolean saveConfigElement() {
        return false;
    }

    @Override
    public Object getCurrentValue() {
        return null;
    }

    @Override
    public Object[] getCurrentValues() {
        return null;
    }
}